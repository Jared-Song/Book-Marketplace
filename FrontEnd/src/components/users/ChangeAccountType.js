import { Button, Grid, TextField, Typography } from "@material-ui/core";
import React from "react";
import { Controller, useForm } from "react-hook-form";
import { makeStyles } from "@material-ui/core/styles";
import { useSnackbar } from "notistack";
import axios from "axios";

const useStyles = makeStyles((theme) => ({
  inputContainer: {
    display: "flex",
    width: 500,
    justifyContent: "space-between",
  },
  inputLabel: {
    position: "relative",
    top: 15,
    whiteSpace: "nowrap",
    paddingRight: theme.spacing(4),
  },
  inputField: {
    width: 250,
  },
  buttonContainer: {
    paddingTop: theme.spacing(5),
  },
}));

export default function ChangeAccountType({ token, user }) {
  const { control, handleSubmit } = useForm();
  const { enqueueSnackbar } = useSnackbar();
  const classes = useStyles();

  const onSubmit = async ({ companyName, abn }) => {
    if (!companyName || !abn) {
      enqueueSnackbar("Company Name or ABN can't be empty", {
        variant: "error",
      });
      return;
    }
    try {
      const { status } = await axios.post(
        process.env.NEXT_PUBLIC_EDIT_USER_URL + user.id,
        {
          ...user,
          business,
          abn,
        },
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        }
      );
      enqueueSnackbar(`Your request has been submitted`, {
        variant: "success",
      });
    } catch (error) {
      enqueueSnackbar("Something is wrong!!", {
        variant: "error",
      });
    }
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <Grid container spacing={2}>
        <Grid item xs={12}>
          <Typography variant="h4">Change User Type</Typography>
        </Grid>
        <Grid item xs={12}>
          <Grid
            container
            direction="column"
            justifyContent="center"
            alignItems="center"
          >
            <Grid item className={classes.inputContainer}>
              <Typography className={classes.inputLabel}>
                Company Name
              </Typography>
              <Controller
                name="companyName"
                control={control}
                render={({ field }) => {
                  return (
                    <TextField
                      {...field}
                      className={classes.inputField}
                      variant="outlined"
                      fullWidth
                      margin="dense"
                    />
                  );
                }}
              />
            </Grid>
            <Grid item className={classes.inputContainer}>
              <Typography className={classes.inputLabel}>ABN</Typography>
              <Controller
                name="abn"
                control={control}
                render={({ field }) => {
                  return (
                    <TextField
                      {...field}
                      variant="outlined"
                      className={classes.inputField}
                      fullWidth
                      margin="dense"
                    />
                  );
                }}
              />
            </Grid>
            <Grid item className={classes.buttonContainer}>
              <Button
                variant="contained"
                color="primary"
                size="large"
                type="submit"
              >
                submit request
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </form>
  );
}
