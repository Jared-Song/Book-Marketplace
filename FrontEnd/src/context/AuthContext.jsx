import React from "react";
import { useJwt } from "react-jwt";

const AuthContext = React.createContext(null);

export const AuthProvider = ({ token, children, loading }) => {
  const [currentToken, setToken] = React.useState(token);
  const { decodedToken } = useJwt(currentToken);
  const [isLoading, setIsLoading] = React.useState(loading);
  React.useEffect(() => {
    setToken(token);
  }, [token]);
  return (
    <AuthContext.Provider
      value={{ currentUser: decodedToken, setToken, isLoading }}
    >
      {children}
    </AuthContext.Provider>
  );
};

export const useCurrentUser = () => React.useContext(AuthContext);
