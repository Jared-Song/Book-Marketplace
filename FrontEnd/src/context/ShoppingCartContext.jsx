import _, { set } from "lodash";
import React from "react";

const ShoppingCartContext = React.createContext(null);

export const ShoppingCartProvider = ({ children }) => {
  const storageItem = typeof window !== "undefined" ? localStorage.getItem("cart") : undefined;
  const [items, setItems] = React.useState(storageItem ? JSON.parse(storageItem) : []);
  React.useEffect(() => {
    localStorage.setItem("cart", JSON.stringify(items));
  }, [items]);
  const removeAllItems = () => {
    setItems([])
    localStorage.clear()
  }

  const removeShoppingCartItem = (id) => {
    const updatedItems = _.remove(items, (item) => {
      return item.id !== id
    })
    setItems(updatedItems)
  }
  const editShoppingCartItem = (newItem) => {
    const updatedItems = items.map((item) => {
      if (newItem.id === item.id) {
        return newItem; 
      }
      return item
    })
    setItems(updatedItems);
  }
  const addIntoShoppingCart = (newItem) => {
    const itemExist = items.filter((item) => item.id === newItem.id);
    if (itemExist.length > 0) {
      const updatedItems = items.map((item) => {
        if (item.id === newItem.id) {
          return {
            ...item,
            quantity: item.quantity + parseInt(newItem.quantity)
          }
        }
        return item;
      })
      setItems(updatedItems);
    } else {
      setItems([
        ...items,
        {
          ...newItem,
          quantity: parseInt(newItem.quantity),
        },
      ]);
    }
  }
  return (
    <ShoppingCartContext.Provider
      value={{
        cartItems: items,
        addIntoShoppingCart,
        editShoppingCartItem,
        removeShoppingCartItem,
        removeAllItems,
      }}
    >
      {children}
    </ShoppingCartContext.Provider>
  );
}

export const useShoppingCart = () => React.useContext(ShoppingCartContext);