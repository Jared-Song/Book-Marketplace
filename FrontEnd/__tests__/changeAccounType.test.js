import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { AuthProvider } from "../src/context/AuthContext";
import { useRouter } from "next/router";
import ChangeAccountType from "../src/components/users/ChangeAccountType";
import { SnackbarProvider } from "notistack";

it("renders Edit Password correctly", () => {
  const tree = renderer
    .create(
      <SnackbarProvider maxSnack={3}>
        <AuthProvider token="testToken">
          <ChangeAccountType
            token="testToken"
            user={{
              id: "1",
              address: "address",
              username: "username",
              fullName: "fullName",
              role: "ADMIN",
              rating: 0,
              ratingNo: 0,
              userStatus: "userStatus",
            }}
          />
        </AuthProvider>
      </SnackbarProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
