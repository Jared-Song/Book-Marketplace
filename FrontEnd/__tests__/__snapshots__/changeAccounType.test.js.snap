// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Edit Password correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
        Change User Type
      </h4>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-inputContainer-1 MuiGrid-item"
        >
          <p
            className="MuiTypography-root makeStyles-inputLabel-2 MuiTypography-body1"
          >
            Company Name
          </p>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-inputField-3 MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                name="companyName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-6"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-inputContainer-1 MuiGrid-item"
        >
          <p
            className="MuiTypography-root makeStyles-inputLabel-2 MuiTypography-body1"
          >
            ABN
          </p>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-inputField-3 MuiFormControl-marginDense MuiFormControl-fullWidth"
          >
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                disabled={false}
                name="abn"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                style={
                  Object {
                    "paddingLeft": 8,
                  }
                }
              >
                <legend
                  className="PrivateNotchedOutline-legend-6"
                  style={
                    Object {
                      "width": 0.01,
                    }
                  }
                >
                  <span
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "&#8203;",
                      }
                    }
                  />
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-buttonContainer-4 MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              submit request
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
